src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SearchScope.cs

```diff
@@ -112,11 +160,17 @@ public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
                     _metadataReference);
             }
 
-            protected override Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
+            protected override async Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
             {
-                return SymbolFinder.FindDeclarationsAsync(
-                    _solution, _assembly, _metadataReference, searchQuery, filter, cancellationToken);
+                var service = _solution.Workspace.Services.GetService<ISymbolTreeInfoCacheService>();
+                var info = await service.TryGetSymbolTreeInfoAsync(_solution, _assembly, _metadataReference, cancellationToken).ConfigureAwait(false);
+                if (info == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ISymbol>();
```

Comment:
Can you doc what this means? Is this because it's not cached yet, might it work later, etc.